Parsers/ASimAuthentication/ProductParsers/vimAuthenticationPostgreSQL copy.yaml

```diff
@@ -0,0 +1,205 @@
+Parser:
+  Title: ASIM PostgreSQL authentication
+  Version: '0.1.1'
+  LastUpdated: April 20, 2022
+Product:
+  Name: PostgreSQL
+Normalization:
+  Schema: Authentication
+  Version: '0.1.1'
+References:
+- Title: ASIM Authentication Schema
+  Link: https://aka.ms/ASimAuthenticationDoc
+- Title: ASIM
+  Link: https:/aka.ms/AboutASIM
+Description: |
+      This Query Parser maps PostgreSQL sign in logs to the Advanced Security Information Model authenticaion schema.
+ParserName: ASimAuthenticationPostgreSQL
+ParserParams:
+  - Name: disabled
+    Type: bool
+    Default: false
+ParserQuery: |
+  let PostgreSQLSignInAuthorized=(starttime:datetime=datetime(null), endtime:datetime=datetime(null), targetusername_has:string="*", disabled:bool=false){
+  PostgreSQL_CL | where not(disabled)
+  // ************************************************************************* 
+  //       <Prefilterring>
+  // *************************************************************************
+  | where 
+    (isnull(starttime)   or TimeGenerated >= starttime)
+    and (isnull(endtime)     or TimeGenerated <= starttime)
+  // ************************************************************************* 
+  //       </Prefilterring>
+  // ************************************************************************* 
+    | where RawData has 'connection authorized'
+    | extend
+    EventVendor = 'PostgreSQL'
+    , EventProduct = 'PostgreSQL'
+    , EventCount = int(1)
+    , EventSchema = 'Authentication'
+    , EventSchemaVersion = '0.1.1'
+    , EventResult = 'Success'
+    , EventStartTime = TimeGenerated
+    , EventEndTime = TimeGenerated
+    , EventType = 'Logon'
+    , DvcHostname = Computer
+    , DvcIpAddr = extract(@'\d{1,3}\.\d{1.3}\.\d{1,3}\.\d{1,3}', 1, Computer)
+    , TargetUsernameType = 'Simple'
+    , TargetUsername = extract(@'user=(.*?)\sdatabase', 1, RawData)
+    , EventResultDetails = 'Connection authorized'
+  // ********************** <Postfilterring> **********************************
+    | where (targetusername_has=='*' or TargetUsername has targetusername_has)
```

Comment:
As part of the prefiltering you need to check if the username appears in the RawData field. The postfiltering than makes sure that this was indeed the username.

Reply:
fixed